tasks/base64-encode-checksum/base64-encode-checksum.yaml

```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: base64-encode-checksum
+  labels:
+    app.kubernetes.io/version: "1.0.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to encode the Terraform Checksum file
+  params:
+    - name: binaries_dir
+      type: string
+      description: Path where the binaries are stored in the workspace
+      default: "binaries"
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used
+      type: string
+      default: ""
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot is stored. The extracted binaries will be stored here as well.
+  results:
+    - name: blob
+      type: string
+      description: blob to sign, that is the result of the base64 encoded checksum
+  steps:
+    - name: base64-encode-checksum
+      image:
+        quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/usr/bin/env sh
+        set -ex
+
+        FILE_PATH="$(workspaces.data.path)/$(params.subdirectory)/$(params.binaries_dir)"
+        base64 -w 0 "$FILE_PATH"/*SHA256SUMS | tee "$(results.blob.path)"
```

Comment:
But that would mean giving the `extract-binaries-from-image` task two different functionalities. Maybe we need in  the future to extract some binaries but not base64 encode anything. Or maybe we would need to base64 encode something without extracting anything and we can only make this task more generic without modifying the other one

Reply:
I took a look at the changes again to understand why we even need to encode it. And I'm not sure. This will be passed to the `sign-checksum-file` which doesn't exist yet. But why does it need the file to be base64 encoded? Won't the task decode it anyway? I'd expect that what we'll want to sign is probably a sha sum of the checksum file, no?

If the `sign-checksum-file` task needs the original checksum file, why not pass it via workspace? That's a better fit for files usually - we switched to that in most places (e.g. collect-data). The main concern when passing data via results is its size - there is a 4k size limit. But if that's not a concern, we could just pass the whole file's contents directly - why bother with base64? If anything, it will make it even bigger.

Reply:
I need to base64 encode the checksum file to pass it as parameter to the internal request, which is going to trigger the internal pipeline and send that blob to sign it. That's why I created this task. All the signing stuff is still WIP so maybe this task could change. If you prefer, I can comment it in the meantime that's finished

Reply:
OK, thanks. I still think this could be easily done in the signing task, but I guess it's fine to do it separately as well.